### PR TITLE
Space Dolphins from YogStation two

### PR DESCRIPTION
**Pull request by Cenrus**
See #997

## About this pull request

dolphins

## Why it's good for the game

dolphins

:cl: Altoids1
imageadd: Space dolphin sprites.
add: Space dolphins and a new event which causes them to spawn in pods around the station.
/:cl:

### CODE REVIEW

**Comment by Baldir-Odensen** [comment]
Are they hostile to crabs?

**Comment by Cenrus** [comment]
No, because crabs are good

**Comment by Wilchenx** [comment]
@Kmc2000 please kill this sprite

**Comment by boodaliboo** [comment]
not nice credit the original pr please https://github.com/yogstation13/Yogstation/pull/2087

**Comment by ike709** [comment]
@boodaliboo Well he did in the original PR that he closed, at least. Which he links in this PR.

**Comment by ike709** [comment]
Also he gave credit to Altoids directly in the CL as the author of the change.

**Comment by eLeCtrOssSnake** [comment]
Uhhhhh, what? Dolphins?

**Comment by eLeCtrOssSnake** [comment]
Space Carp, now we have Space Dolphins uh, ok.

**Comment by eLeCtrOssSnake** [comment]
But where did they come from?

**Comment by boodaliboo** [comment]
> Also he gave credit to Altoids directly in the CL as the author of the change.

i know im just having fun

File: code/modules/mob/living/simple_animal/hostile/retaliate/dolphin.dm [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+	//Space dolphins aren't affected by cold.
+	atmos_requirements = list("min_oxy" = 0, "max_oxy" = 0, "min_tox" = 0, "max_tox" = 0, "min_co2" = 0, "max_co2" = 0, "min_n2" = 0, "max_n2" = 0)
+	minbodytemp = 0
+	maxbodytemp = 1500
+
+	faction = list("dolphin")//why do carps not attack dolphins again? //i dunno now they do
+
+/mob/living/simple_animal/hostile/retaliate/dolphin/Process_Spacemove(movement_dir = 0)
+	return TRUE	//No drifting in space for space dolphins!	//original comments do not steal //Too late.
```
Comment (eLeCtrOssSnake) [inline code/modules/mob/living/simple_animal/hostile/retaliate/dolphin.dm]: What are these comments though? I think they aren't required

File: code/modules/events/dolphin_migration.dm [hunk 2]
```diff
@@ -0,0 +1,22 @@
+//DOLPHIN PORT WOOOOOO
```
Comment (eLeCtrOssSnake) [inline code/modules/events/dolphin_migration.dm]: I think this ain't required too.

File: code/modules/mob/living/simple_animal/hostile/retaliate/dolphin.dm [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+	maxHealth = 25
+	health = 25
+	a_intent = "harm"
+	spacewalk = TRUE
+
+	environment_smash = 0
+	harm_intent_damage = 8
+	melee_damage_lower = 15
+	melee_damage_upper = 15
+	pass_flags = PASSTABLE //secretly a tactical dolphin buff.
```
Comment (eLeCtrOssSnake) [inline code/modules/mob/living/simple_animal/hostile/retaliate/dolphin.dm]: This is useless too.

**Review by eLeCtrOssSnake — CHANGES_REQUESTED** [review]
Express your changes in a half-formal way, not "fun stuff don't touch"

File: code/modules/events/dolphin_migration.dm [hunk 4]
```diff
@@ -0,0 +1,22 @@
+//DOLPHIN PORT WOOOOOO
+/datum/round_event_control/dolphin_migration
+	name = "Dolphin Migration"
+	typepath = /datum/round_event/dolphin_migration
+	weight = 15
+	earliest_start = 10 MINUTES
+	max_occurrences = 6
+
+/datum/round_event/dolphin_migration
+	announceWhen	= 3
```
Comment (ike709) [inline code/modules/events/dolphin_migration.dm]: Bad whitespace

File: code/modules/mob/living/simple_animal/hostile/retaliate/dolphin.dm [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+/mob/living/simple_animal/hostile/retaliate/dolphin/AttackingTarget()
+	..()
+	if(ishuman(target))
+		var/mob/living/carbon/human/H = target
+		H.adjustStaminaLoss(8)
+
+/mob/living/simple_animal/hostile/retaliate/dolphin/handle_automated_action()
+	if(..())
+		for(var/mob/living/simple_animal/hostile/carp/C in view(src, 10))
```
Comment (ike709) [inline code/modules/mob/living/simple_animal/hostile/retaliate/dolphin.dm]: 🤔

File: code/modules/mob/living/simple_animal/hostile/retaliate/dolphin.dm [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+	attacktext = "bites"
+	attack_sound = 'sound/weapons/bite.ogg'
+	speak_emote = list("chitters", "squeeks", "clicks")
+
+	//Space dolphins aren't affected by cold.
+	atmos_requirements = list("min_oxy" = 0, "max_oxy" = 0, "min_tox" = 0, "max_tox" = 0, "min_co2" = 0, "max_co2" = 0, "min_n2" = 0, "max_n2" = 0)
+	minbodytemp = 0
+	maxbodytemp = 1500
+
+	faction = list("dolphin")//why do carps not attack dolphins again? //i dunno now they do
```
Comment (ike709) [inline code/modules/mob/living/simple_animal/hostile/retaliate/dolphin.dm]: So dolphins attack carp but carp don't attack dolphins...?

**Comment by Cenrus** [comment]
Ready now.

**Pull request merged**